### DISCUSSION TITLE
GraphQL Queries for property comparison between different nodes

### DISCUSSION THREAD

**Discussion by Palak-K9** [neo4j-graphql/neo4j-graphql-java#319] [Q&A]
I am trying to **convert** some **Cypher Queries** to **GraphQL queries** using _neo4j-graphql-java_

I have the following nodes - Application, Customer, PremCustomer, Channel

I am using the below Schema

```
type Application
 Id: Int!
 author: String!
 code: Int!
 name: String!
 hasCustomers: [Customer!]! @relation(name: "HAS_CUSTOMER", direction: OUT)
}

type Customer{
 custId: Int!
 premId: Int!
 name: String!
 value: Int!
}

type PremCustomer{
 Id: Int!
 name: String!
}

type Channel{
 Id: Int
 name: String!
 rating: Int!
}
```

I have the following Cypher queries - 

```
MATCH (app: Application)
OPTIONAL MATCH (app)-[:HAS_CUSTOMER]->(cust: Customer {premId: 5}) 
OPTIONAL MATCH (premCust:PremCustomer) WHERE premCust.Id=cust.premId 
OPTIONAL MATCH (ch: Channel) WHERE ch.Id=cust.value
RETURN DISTINCT app{a: premCust.Id, b: premCust.name, c: cust.value, d: ch.rating}
```

How can I create Graphql query/ies for the above Cypher Query?

I tried to do **Id comparisons** for different nodes but the syntax doesnt allow this.
I also tried the below syntax - 

```
{
  firstPart: applications(where: {hasCustomers_some: {premId: 5}}){
       hasCustomers{
           premId
           name
       }
 }
```
 
 And then tried to utilise the **alias** _firstPart_ to use its results for further comparison like below-
 
 ```
 premCustomers(where: {Id=firstPart.hasCustomers.premId}){
 ...
 }
 
 But it isn't working.
 ```
 
 So, Is there any other syntax for GraphQL to achieve the results I am getting using the Cypher Queries?

**Comment by Andy2003** [comment]
With a singe request it is not possible to achieve with your schema.
Your schema does not have any relation between `Customer` and `PremCustomer`.
For me it looks like you  are trying to use the ids as foreign key as in an RDBMS. Have you tried to model your schema with relations?

**Reply by Palak-K9** [reply]
Yes, my schema does not have any relation between Customer and PremCustomer, actually, the requirement is like this.

_With a **single request** it is not possible to achieve with your schema._  - Can we achieve the desired result using multiple queries? (Condition: no relation between Customer and PremCustomer)
